toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1434,3 +1574,64 @@ func findAllRPMS(rpmsToFind []string) (foundAllRpms bool, missingRpms []string)
 
 	return
 }
+
+// findAllRPMSUpstream returns true if all RPMs requested are found in PMC/repo.
+//  This is used to check if all RPMs are available in PMC/repo before starting a build.
+//  needs repo-list, manifest files rpmrepocloner
+func findAllRPMSUpstream(rpmsToFind []*PkgNode, outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir string, usePreviewRepo, ignoreVersionToResolveSelfDep bool, repoFiles []string) (foundAllRpms bool) {
+
+        // Create the worker environment
+// Initialize initializes rpmrepocloner, enabling Clone() to be called.
+//   - destinationDir is the directory to save RPMs
+//   - tmpDir is the directory to create a chroot
+//   - workerTar is the path to the worker tar used to seed the chroot
+//   - existingRpmsDir is the directory with prebuilt RPMs
+//   - prebuiltRpmsDir is the directory with toolchain RPMs
+//   - usePreviewRepo if set, the upstream preview repository will be used.
+//   - repoDefinitions is a list of repo files to use when cloning RPMs
+
+        cloner := rpmrepocloner.New()
+	err := cloner.Initialize(outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir, usePreviewRepo, repoFiles)
+        if err != nil {
+                logger.Log.Errorf("Failed to initialize RPM repo cloner. Error: %s", err)
+                return
+        }
+        defer cloner.Close()
+
+        err = cloner.AddNetworkFiles("", "")
+        if err != nil {
+                logger.Log.Panicf("Failed to customize RPM repo cloner. Error: %s", err)
```

Comment:
>ogger.Log.Panicf("Failed to customize RPM repo cloner. Error: %s", err) [](http://example.com/codeflow?start=17&length=71)

Please never panic from internal code. Return an error and leave the decision to the top-level caller.